migrations/04_singleton_proxy.js

```diff
@@ -0,0 +1,104 @@
+const ethUtils = require('ethereumjs-util')
+const fs = require('fs')
+const utils = require('../utils/utils.js')
+
+const SingletonFactory = artifacts.require("SingletonFactory")
+const WitnetRequestBoard = artifacts.require("WitnetRequestBoard")
+
+module.exports = async function (deployer, network, accounts) {
+
+  let addresses = require('./addresses.json')
+
+  const factory = await SingletonFactory.deployed()
+  const wrb = await WitnetRequestBoard.deployed()
+
+  const singletons = require('./singletons.json') 
+  const artifact = artifacts.require("WitnetRequestBoardProxy")
+  const contract = artifact.contractName
+
+  if (singletons.contracts[contract]) {
+
+    let addresses = require('./addresses.json')
+
+    const from = singletons.from
+        ? singletons.from
+        : deployer.networks[network].from
+          ? deployer.networks[network].from
+          : accounts[0]
+      ;
```

Comment:
Same as above, this may be easier to reason about:
```suggestion
    const from = singletons.from || deployer.networks[network].from || accounts[0]
```

Reply:
Yep.